ci/e2e-tests/helper-functions.sh

```diff
@@ -0,0 +1,179 @@
+# assigns devices to a 'foo-devices' IoT hub if they have a payload containing
+# a 'modelId' field with 'foo' in it. 
+function setupCustomAllocationPolicy {
+        echo 'Creating an Azure Function for use as a DPS custom allocation policy...' >&2
+        
+        # Initialize function app project and function
+        func init DpsCustomAllocationFunctionProj --dotnet
+        pushd DpsCustomAllocationFunctionProj
+        func new --name "$dps_allocation_function_name" --template "HTTP trigger" --authlevel "anonymous" --language C# --force
+        
+        # Copy source code into functino app project
+        cp "$GITHUB_WORKSPACE/ci/e2e-tests/DpsCustomAllocation.cs" .
+
+        # Add build deps
+        dotnet add package Microsoft.Azure.Devices.Provisioning.Service -v 1.16.3
+        dotnet add package Microsoft.Azure.Devices.Shared -v 1.27.0
+
+        # Create storeage account needed by function app
+        sa_name="sa${GITHUB_RUN_ID}r${GITHUB_RUN_NUMBER}"
+        az storage account create \
+            --name $sa_name \
+            --location $AZURE_LOCATION \
+            --resource-group $AZURE_RESOURCE_GROUP_NAME \
+            --sku Standard_LRS \
+            --tags "suite_id=$suite_id"
+
+        # Create function app
+        az functionapp create \
+            --resource-group $AZURE_RESOURCE_GROUP_NAME \
+            --consumption-plan-location $AZURE_LOCATION \
+            --runtime dotnet \
+            --functions-version 3 \
+            --name "$dps_allocation_functionapp_name" \
+            --disable-app-insights \
+            --storage-account $sa_name \
+            --tags "suite_id=$suite_id"
+
+        # Publishing the app sometimes fails, so retry up to 3 times
+        set +e
+        for retry in {0..3}; do
+            if [ "$retry" != "0" ]; then
+                sleep 10
+            fi
+            echo "Publishing the function app. Attempt $retry..."
+            func azure functionapp publish "$dps_allocation_functionapp_name" --force
+            if [ "$?" == "0" ]; then
+                break
+            fi
+            if [ "$retry" == "3" ]; then
+                exit 1
+            fi
+        done
+        set -e
+        
+        popd
+        echo 'Created an Azure Function for use as a DPS custom allocation policy.' >&2
+
+        echo 'Creating a second IoT hub for testing the custom allocation policy...'
+        createHub "$foo_devices_iot_hub"
+        dps_resource_id=$(az iot dps show \
+            --name $suite_common_resource_name \
+            --resource-group $AZURE_RESOURCE_GROUP_NAME \
+                | jq '.id' -r)
+        createDpsLinkedHub $suite_common_resource_name "$foo_devices_iot_hub" $dps_resource_id "suite_id=$suite_id"
+        echo 'Created second IoT hub.'
+}
+
+# Install tools needed to run the tests
+function installTestTools {
+    echo 'Installing test tools...' >&2
+    distributor_id=$(lsb_release -is)
```

Comment:
```suggestion
    distributor_id="$(lsb_release -is)"
```

And similar for other assignments.

Reply:
Done.